apistrap/aiohttp.py

```diff
@@ -0,0 +1,221 @@
+from itertools import chain
+
+import json
+import logging
+import mimetypes
+from aiohttp import web
+from aiohttp.web_exceptions import HTTPError
+from aiohttp.web_request import BaseRequest
+from aiohttp.web_response import Response
+from pathlib import Path
+from schematics import Model
+from schematics.exceptions import DataError
+from typing import Type, Optional, Coroutine, Callable, Tuple, Any
+
+from apistrap import Swagger
+from apistrap.decorators import RespondsWithDecorator, AcceptsDecorator
+from apistrap.errors import UnexpectedResponseError, InvalidResponseError, ApiClientError
+from apistrap.schemas import ErrorResponse
+from apistrap.types import FileResponse
+from apistrap.utils import format_exception
+
+
+class AioHTTPRespondsWithDecorator(RespondsWithDecorator):
+    async def _process_response(self, response, is_last_decorator: bool, *args, **kwargs):
+        if isinstance(response, Response) or isinstance(response, web.StreamResponse):
+            return response
+        if isinstance(response, FileResponse):
+            # TODO consider implementing add_etags, cache_timeout and conditional
+            headers = {}
+
+            if self._mimetype:
+                headers['Content-Type'] = self._mimetype
+            elif response.mimetype:
+                headers['Content-Type'] = response.mimetype
+            elif response.attachment_filename:
+                headers['Content-Type'] = mimetypes.guess_type(response.attachment_filename)[0]
+
+            if response.last_modified is not None:
+                headers['Last-Modified'] = response.last_modified
+
+            if response.as_attachment:
+                if response.attachment_filename is None:
+                    raise TypeError('Missing attachment filename')
+
+                headers['Content-Disposition'] = f'attachment,filename={response.attachment_filename}'
+
+            if isinstance(response.filename_or_fp, str) or isinstance(response.filename_or_fp, Path):
+                return web.FileResponse(response.filename_or_fp, headers=headers)
+            else:
+                stream = web.StreamResponse(headers=headers)
+                request = next(filter(lambda a: isinstance(a, BaseRequest), chain(args, kwargs.values())), None)
+
+                if request is None:
+                    raise TypeError('No request passed to view function')
+
+                await stream.prepare(request)
```

Comment:
this is nice! `aiohttp` seems to be pretty cool

Reply:
It is! It also supports websockets and other means of async communication, which might be useful later.